### DISCUSSION TITLE
The time of the alarm field cannot be adjusted according to the timezone

### DISCUSSION THREAD

**Discussion by ixiaoyi93** [jertel/elastalert2#701] [Q&A] [answered]
elastalert2 is a very good open source project, thank you to the developers for your efforts.

However, in the process of using it, I encountered a major headache. elasticsearch stores the time zone in UTC by default, and the time zone of the alerts sent via elastalter2 is also UTC, which does not convert the time to local time.

Here are the times stored by elasticsearch.
![image](https://user-images.githubusercontent.com/34268642/153133383-c6fb85d3-7e1d-4a1c-9a3f-250dfb3aa3a0.png)

The following alerts are sent.
![image](https://user-images.githubusercontent.com/34268642/153133427-921a5d88-f5bd-4739-bd99-af74f897952b.png)

elastalert2 rule configuration.
```
name:  pay online error 

type: any

use_strftime_index: true
index: k8s-g7pay-logs-%Y.%m.%d

query_timezone: "Asia/Shanghai"

include: ["@timestamp","level","project_name","env","msg","time", "log_id"]

filter:
- query:
    query_string:
      query: "(project_name:service-bank-notify  AND level: error"


# FeishuAlert
alert:
  - "elastalert_modules.feishu_alert.FeishuAlert"

feishualert_url: "https://open.feishu.cn/open-apis/bot/v2/hook/"

# 飞书机器人id
feishualert_botid:
  ""

# 告警标题
feishualert_title:
  "支付生产错误日志告警"

# 这个时间段内的匹配将不告警，适用于某些时间段请求低谷避免误报警
feishualert_skip:
  start: "00:00:00"
  end: "00:00:00"

# 告警内容定制，可以直接引用 include 里面定义ES索引的字段
feishualert_body:
  "
  事件名称:  支付生产错误日志告警\n
  日志 ID:  log_id: {log_id}\n
  触发时间:  time:  {time}\n
  应用名称:  project_name: {project_name}\n
  应用环境:  env: {env}\n
  日志级别:  level: {level}\n
  告警内容:  msg: {msg}
  "
```

elastalert2 version:
```
elastalert2 version: 2.2.3
elastalert2-server version: 5.0.0-next.2

```

**Comment by nsano-rururu** [comment, marked as answer]
https://stackoverflow.com/questions/39763392/how-to-adjust-timestamp-to-local-time-zone-in-elastalert

**Reply by nsano-rururu** [reply]
https://kenchii.blog/elastalert-timestamp-convert-to-jst/

**Reply by nsano-rururu** [reply]
https://github.com/Yelp/elastalert/issues/861

**Reply by nsano-rururu** [reply]
https://github.com/bitsensor/elastalert/issues/204

**Reply by nsano-rururu** [reply]
https://elastalert2.readthedocs.io/en/latest/elastalert.html#configuration
>custom_pretty_ts_format: This option provides a way to define custom format of timestamps printed in log messages and in alert messages. If this option is not set, default timestamp format (‘%Y-%m-%d %H:%M %Z’) will be used. (Optional, string, default None)
